packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-streaming-job.ts

```diff
@@ -0,0 +1,181 @@
+/**
+ *  Python Spark Streaming Jobs class
+ *
+ * A Streaming job is similar to an ETL job, except that it performs ETL on data streams
+ * using the Apache Spark Structured Streaming framework.
+ * These jobs will default to use Python 3.9.
+ *
+ * Similar to ETL jobs, streaming job supports Scala and Python languages. Similar to ETL,
+ * it supports G1 and G2 worker type and 2.0, 3.0 and 4.0 version. We’ll default to G2 worker
+ * and 4.0 version for streaming jobs which developers can override.
+ * We will enable —enable-metrics, —enable-spark-ui, —enable-continuous-cloudwatch-log.
+ *
+ * RFC : https://github.com/aws/aws-cdk-rfcs/blob/main/text/0497-glue-l2-construct.md
+ */
+
+import { CfnJob } from 'aws-cdk-lib/aws-glue';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import { Bucket } from 'aws-cdk-lib/aws-s3';
+import { Job, JobProperties } from './job';
+import { Construct } from 'constructs';
+import { JobType, GlueVersion, JobLanguage, PythonVersion, WorkerType } from '../constants';
+import { SparkUIProps, SparkUILoggingLocation, validateSparkUiPrefix, cleanSparkUiPrefixForGrant } from './spark-ui-utils';
+
+/**
+ * Properties for creating a Python Spark ETL job
+ */
+export interface PySparkStreamingJobProps extends JobProperties {
+
+  /**
+   * Enables the Spark UI debugging and monitoring with the specified props.
+   *
+   * @default - Spark UI debugging and monitoring is disabled.
+   *
+   * @see https://docs.aws.amazon.com/glue/latest/dg/monitor-spark-ui-jobs.html
+   * @see https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html
+   */
+  readonly sparkUI?: SparkUIProps;
+
+  /**
+   * Extra Python Files S3 URL (optional)
+   * S3 URL where additional python dependencies are located
+   * @default - no extra files
+  */
+  readonly extraPythonFiles?: string[];
+}
+
+/**
+ * A Python Spark Streaming Glue Job
+ */
+export class PySparkStreamingJob extends Job {
```

Comment:
What is the benefit to create a separate class for Streaming? It seems there are many duplicate code.

Reply:
The goal for the re-architecture of the Glue L2 construct is to enforce defaults and parameters via contracts and interfaces rather than the sync-time validations that currently exist in the construct. Having separate classes allow us to inherit the common aspects of the jobs, but enforce the particular differences via the job-specific interfaces.

The end goal is to allow for a better developer experience making it absolutely clear what's required and what's not needed / won't be used at development time rather than at synth time.